ddtrace/tracer/textmap_test.go

```diff
@@ -355,210 +355,244 @@ func TestTextMapPropagator(t *testing.T) {
 	})
 }
 
-func testB3(t *testing.T, b3Header string) {
-	t.Run("inject", func(t *testing.T) {
-		os.Setenv("DD_PROPAGATION_STYLE_INJECT", b3Header)
-		defer os.Unsetenv("DD_PROPAGATION_STYLE_INJECT")
-
-		var tests = []struct {
-			in  []uint64
-			out map[string]string
-		}{
-			{
-				[]uint64{1412508178991881, 1842642739201064},
-				map[string]string{
-					b3TraceIDHeader: "000504ab30404b09",
-					b3SpanIDHeader:  "00068bdfb1eb0428",
+func TestEnvVars(t *testing.T) {
+	var testEnvs []map[string]string
+
+	testEnvs = []map[string]string{
+		{headerPropagationStyleInject: "b3"},
+		{headerPropagationStyleInjectDeprecated: "b3,none" /* none should have no affect */},
+		{headerPropagationStyle: "b3"},
+		{headerPropagationStyleInject: "b3multi", headerPropagationStyleInjectDeprecated: "none" /* none should have no affect */},
+		{headerPropagationStyleInject: "b3multi", headerPropagationStyle: "none" /* none should have no affect */},
+	}
+	for _, testEnv := range testEnvs {
+		t.Run(fmt.Sprintf("inject with env=%q", testEnv), func(t *testing.T) {
+			for k, v := range testEnv {
+				os.Setenv(k, v)
+				defer os.Unsetenv(k)
+			}
+			var tests = []struct {
+				in  []uint64
+				out map[string]string
+			}{
+				{
+					[]uint64{1412508178991881, 1842642739201064},
+					map[string]string{
+						b3TraceIDHeader: "000504ab30404b09",
+						b3SpanIDHeader:  "00068bdfb1eb0428",
+					},
 				},
-			},
-			{
-				[]uint64{9530669991610245, 9455715668862222},
-				map[string]string{
-					b3TraceIDHeader: "0021dc1807524785",
-					b3SpanIDHeader:  "002197ec5d8a250e",
+				{
+					[]uint64{9530669991610245, 9455715668862222},
+					map[string]string{
+						b3TraceIDHeader: "0021dc1807524785",
+						b3SpanIDHeader:  "002197ec5d8a250e",
+					},
 				},
-			},
-			{
-				[]uint64{1, 1},
-				map[string]string{
-					b3TraceIDHeader: "0000000000000001",
-					b3SpanIDHeader:  "0000000000000001",
+				{
+					[]uint64{1, 1},
+					map[string]string{
+						b3TraceIDHeader: "0000000000000001",
+						b3SpanIDHeader:  "0000000000000001",
+					},
 				},
-			},
-		}
-
-		for _, test := range tests {
-			t.Run("", func(t *testing.T) {
-				tracer := newTracer()
-				root := tracer.StartSpan("web.request").(*span)
-				root.SetTag(ext.SamplingPriority, -1)
-				root.SetBaggageItem("item", "x")
-				ctx, ok := root.Context().(*spanContext)
-				ctx.traceID = test.in[0]
-				ctx.spanID = test.in[1]
-				headers := TextMapCarrier(map[string]string{})
-				err := tracer.Inject(ctx, headers)
-
-				assert := assert.New(t)
-				assert.True(ok)
-				assert.Nil(err)
-				assert.Equal(test.out[b3TraceIDHeader], headers[b3TraceIDHeader])
-				assert.Equal(test.out[b3SpanIDHeader], headers[b3SpanIDHeader])
-			})
-		}
-	})
+			}
+
+			for _, test := range tests {
+				t.Run("", func(t *testing.T) {
+					tracer := newTracer()
+					root := tracer.StartSpan("web.request").(*span)
+					root.SetTag(ext.SamplingPriority, -1)
+					root.SetBaggageItem("item", "x")
```

Comment:
Not exaclty about env variables, but since we're touching the tests already:
All of the `root.SetBaggageItem("item", "x")` in this file, except in TestTextMapPropagatorInjectHeader, line 139 have no effect. 

The tests either don't check the propagation of the baggage(like in this specific case), or only check its presence but not value. So probably should remove this line or add a check verifying the baggage is as expected where applicable

Reply:
Good catch! Removed those and added a different test that both injects and extracts to verify behavior.